DecipherNow/helm-charts
# Init jobs

Adds back the init jobs to both catalog and control-api to configure the mesh on startup, and fixes the scripts to also succeed on `helm upgrade`.

To test:

1. Bring up the mesh:

( If you want to view the dashboard - make sure voyager is installed:
```bash
helm install voyager-operator appscode/voyager \
--version v12.0.0-rc.0   \
--namespace kube-system   \
--set cloudProvider=minikube
```
)

```bash
helm dep up secrets
helm install secrets secrets -f credentials.yaml
```

```bash
helm dep up fabric
helm install fabric fabric --set=global.environment=kubernetes
```

Notice that the control-api-init job should run and complete successfully.

```bash
helm dep up edge
helm install edge edge --set=global.environment=kubernetes
```

```bash
helm dep up data
helm install data data --set=global.environment=kubernetes
```

```bash
helm dep up sense
helm install sense sense  --set=global.environment=kubernetes
```

Notice that the catalog-init job should run and complete successfully.

2. If you have the new quickstart certs imported into your browser, you should now be able to access the dashboard or use the greymatter cli and see that catalog and the mesh have been configured.

3.  Test helm upgrades:

```bash
helm upgrade fabric fabric --set=global.environment=kubernetes
```
The control-api-init job should again complete successfully.

```bash
helm upgrade sense sense --set=global.environment=kubernetes
```
The catalog-init job should again complete successfully.
